src/pipes/db-lookup.pipe.ts

```diff
@@ -4,11 +4,18 @@ import {
   ValidationArguments,
 } from "class-validator";
 import { FindOneOptions } from "typeorm";
-import { DbLookupValidator } from "../validators/db-lookup.validator";
+import { DbLookupValidator } from '../validators';
 
 export function DbLookup(
   type: any,
-  findOption?: (validationArguments: ValidationArguments) => FindOneOptions,
+  findOption?:
+    | ((validationArguments: ValidationArguments) => FindOneOptions)
+    | "primary"
+    | {
+        validateNumeric?: boolean;
+        ignoreEmpty?: boolean;
```

Comment:
Why is this duplicated here? If it's to provide a keyed object for providing arguments, I think it would be best to refactor the function's parameters to _all_ be in an object. The current way works but I think it could be confusing.

Reply:
``` markdown
Good catch!
Done in the revised version. 
It now accepts a single configuration object, making the API cleaner and removing any parameter duplication.